flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -1181,11 +1195,21 @@ public FileVisitResult visitFile(java.nio.file.Path file, BasicFileAttributes at
 					envShipFileList.append(key).append("=").append(remotePath).append(",");
 
 					// add files to the classpath
-					classPaths.add(key);
+					if (key.endsWith("jar")) {
```

Comment:
and these following 5 lines.

It could be `addFilesToClassPath(classPath, shipFile)`